Purchases/Networking/Operations/LogInOperation.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  LogInOperation.swift
+//
+//  Created by Joshua Liebowitz on 11/19/21.
+
+import Foundation
+
+class LogInOperation: NetworkOperation {
+
+    let httpClient: HTTPClient
+    let authHeaders: [String: String]
+    private let loginCallbackCache: CallbackCache<LogInCallback>
```

Comment:
i can sometimes over-architect generics, but we could maybe think about having a `CachedCallbackNetworkOperation` that extends from `NetworkOperation`, with a generic Callback type and maybe even cacheKey?

Reply:
That's exactly where my head was going too! I wanted to get this big refactor locked down first, then I was going to take another pass at consolidating more of this type of logic. Wanna help?

Reply:
would love to!